app/src/main/java/com/example/calculatorkt/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -30,13 +33,16 @@ class CalculatorPresenter(
         when (model.resultEnum) {
             ResultEnum.DIVIDE_BY_ZERO_ERROR -> view.showDivideByZero()
             ResultEnum.INCOMPLETE_OPERATION_ERROR -> view.showIncompleteOperation()
-            ResultEnum.SUCCESS -> view.showResultValue(result)
+            ResultEnum.SUCCESS -> {
+                view.showResultValue(result)
+                model.equalsPressed = true
+            }
         }
     }
 
     override fun onClearLastButtonPressed() {
         model.deleteLast()
-        if (model.operationValue.equals(EMPTY)) {
+        if (model.operationValue == EMPTY) {
```

Comment:
you can use .isEmpty() instead of the == EMPTY

Reply:
Solved!